public/app/plugins/datasource/cloudwatch/datasource.test.ts

```diff
@@ -225,6 +225,36 @@ describe('datasource', () => {
         expect(response.data.length).toEqual(2);
       });
     });
+
+    it('sets fields.config.interval based on period', async () => {
```

Comment:
could we add  a case for undefined `meta.custom.period`  (either no `meta` or no `custom`) to show that we are happy with the auto behaviour? 

```
            c: {
              refId: 'c',
              series: [{ name: 'memory', points: [1, 2], meta: { } }],
            }
```

Reply:
Hmmm it seems like the `meta.custom.period` won't be assigned here. What I understand is that the period is determined [here](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/cloudwatch/datasource.ts#L482) (from the handler [here](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/cloudwatch/datasource.ts#L276
) which then calls the function `performTimeSeriesQuery` which we're unit testing in this PR). 

If it's possible we can:
* add a test specifically for `getPeriod` in the case of `auto`
* would it be possible to add a test on `handleMetricQueries` which shows how it calls `performTimeSeriesQuery`?

But I guess this could be done in a separate PR (soon).